datavault-webapp/src/main/java/org/datavaultplatform/webapp/security/ScopedPermissionEvaluator.java

```diff
@@ -0,0 +1,112 @@
+package org.datavaultplatform.webapp.security;
+
+import com.google.common.collect.Multimap;
+import org.datavaultplatform.common.model.Group;
+import org.datavaultplatform.common.model.Permission;
+import org.datavaultplatform.common.model.Vault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.PermissionEvaluator;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.TreeMap;
+
+/**
+ * A permission expression evaluator that is able to check existence of {@link Vault} and {@link Group} permissions
+ * on a given authentication.
+ *
+ * <p>
+ * For example, it can be used in a pre-authorize filter:
+ * <pre>
+ *    {@literal @}PreAuthorize("hasPermission(#vault, 'VAULT_PERMISSION_1')")}
+ *     public void deleteVault(Vault vault) {
+ *         ...
+ *     }
+ * </pre>
+ *
+ * <p>
+ * Or called imperatively using any {@link Permission} value.
+ * <pre>
+ *     Vault vault = ...;
+ *     if (permissionEvaluator.hasPermission(user, vault, Permission.VAULT_PERMISSION_1)) { ... }
+ * </pre>
+ */
+public class ScopedPermissionEvaluator implements PermissionEvaluator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ScopedPermissionEvaluator.class);
+
+    /**
+     * Real implementation for actual checking of permissions on an authentication.
+     *
+     * @param auth         The authentication to extract permissions from.
+     * @param key          The unique key representing the scope of the requested permission.
+     * @param permissionId A string or enum value representing the permission.
+     * @return {@code true} iff authorized, {@code false} otherwise.
+     */
+    boolean hasPermission(Authentication auth, ScopedPermissionKey key, Object permissionId) {
+        if (auth == null) {
+            return false;
+        }
+
+        Collection<? extends GrantedAuthority> authorities = auth.getAuthorities();
+        if (authorities.contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
```

Comment:
This role is given to anyone who can see the admin dashboard, even if they're scoped to a single school

Reply:
That shouldn't be the case. `ROLE_ADMIN` is IS admin.

Reply:
Why shouldn't it? `ROLE_ADMIN` used to mean IS admin but now means that the user can see the admin page

Reply:
So how are you supposed to tell if someone is an IS admin? And why would you change it from the existing mechanism?